targets/csharp/make.js

```diff
@@ -263,32 +253,32 @@ function getPropertyCsType(property, datatype, needOptional) {
         throw "Unknown property type: " + property.actualtype + " for " + property.name + " in " + datatype.name;
 }
 
-function getAuthParams(apiCall) {
+function getAuthParams(apiCall, isApiInstance = false) {
     if (apiCall.url === "/Authentication/GetEntityToken")
         return "authKey, authValue";
     if (apiCall.auth === "EntityToken")
         return "\"X-EntityToken\", PlayFabSettings.EntityToken";
-    if (apiCall.auth === "SecretKey")
+    if (apiCall.auth === "SecretKey" && isApiInstance === false)
         return "\"X-SecretKey\", PlayFabSettings.DeveloperSecretKey";
-    if (apiCall.auth === "SecretKeyInstance")
+    if (apiCall.auth === "SecretKey" && isApiInstance === true)
         return "\"X-SecretKey\", developerSecretKey";
```

Comment:
Let's check if developerSecretKey is string.IsNullOrEmpty(), and if it is, then fall back on PlayFabSettings.DeveloperSecretKey.
@Vicodex do you agree?

Reply:
It already generates something like you said. Here is a is a bunch of code which is generated from this part:  
![image](https://user-images.githubusercontent.com/4026209/52616214-ca01a680-2ea8-11e9-8d9e-ef6c01f7679d.png)